ras_party/controllers/account_controller.py

```diff
@@ -347,6 +348,74 @@ def resend_verification_email(party_uuid, session):
     return {'message': EMAIL_VERIFICATION_SENT}
 
 
+@transactional
+@with_db_session
+def add_new_survey_for_respondent(payload, tran, session):
+    """
+    Add a survey for an existing respondent
+    :param payload: json containing party_id and enrolment_code
+    :param session: database session
+    """
+    logger.info("Enrolling existing respondent in survey")
+    v = Validator(Exists('party_id'))
+    if not v.validate(payload):
+        raise RasError(v.errors, 400)
+
+    v1 = Validator(Exists('enrolment_code'))
+    if not v1.validate(payload):
+        raise RasError(v1.errors, 400)
+
+    respondent_party_id = payload['party_id']
+    enrolment_code = payload['enrolment_code']
+
+    iac = request_iac(enrolment_code)
+    if not iac.get('active'):
+        raise RasError("Enrolment code is not active.", status_code=400)
+
+    respondent = query_respondent_by_party_uuid(respondent_party_id, session)
+
+    case_context = request_case(enrolment_code)
+    case_id = case_context['id']
+    business_id = case_context['partyId']
+    collection_exercise_id = case_context['caseGroup']['collectionExerciseId']
+    collection_exercise = request_collection_exercise(collection_exercise_id)
+
+    try:
+        survey_id = collection_exercise['surveyId']
+        survey = request_survey(survey_id)
+        survey_name = survey['longName']
+    except KeyError:
+        raise RasError("There is no survey bound for this user with id: ".format(email_address=respondent.id))
+
+    if query_business_respondent_by_respondent_id_and_business_id(business_id, respondent.id, session) is None:
+        """
+        Associate respondent with new business
+        """
+        business = query_business_by_party_uuid(business_id, session)
+        if not business:
+            msg = f"Could not locate business with id '{business_id}' when creating business association."
+            raise RasError(msg, status_code=404)
+        br = BusinessRespondent(business=business, respondent=respondent)
+    else:
+        br = query_business_respondent_by_respondent_id_and_business_id(business_id, respondent.id, session)
+    pending_enrolement = PendingEnrolment(case_id=case_id,
+                                          respondent=respondent,
+                                          business_id=business_id,
+                                          survey_id=survey_id)
+    Enrolment(business_respondent=br,
+              survey_id=survey_id,
+              survey_name=survey_name,
+              status=EnrolmentStatus.PENDING)
```

Comment:
Depending on if there is a case for a pending enrolment or not this may need to be changed to ENABLED